leshan-core-cf/src/main/java/org/eclipse/leshan/core/californium/oscore/cf/InMemoryOscoreContextDB.java

```diff
@@ -46,6 +46,7 @@ public synchronized OSCoreCtx getContext(byte[] rid, byte[] IDContext) throws Co
 
         if (osCoreCtx == null) {
             OscoreParameters params = store.getOscoreParameters(rid);
+            // TODO OSCORE we should handle if param is null ?
             osCoreCtx = deriveContext(params);
             super.addContext(osCoreCtx);
         }
```

Comment:
If we have not oscore parameters for this RID.
We should not be able to decrypt the payload and reject the message.

What is the right way to handle this ? return `null` ?

Reply:
> If we have not oscore parameters for this RID. We should not be able to decrypt the payload and reject the message.
> 
> What is the right way to handle this ? return `null` ?

In the context of OSCORE, if getContext returns null then the ObjectSecurityLayer will understand that no context was found for that peer and then react to that. So for instance for an incoming request it will respond with a 4.01 (Unauthorized) error message with payload string "Security context not found". So if the OSCORE code in Californium is allowed to handle things, that will be the result of a null from getContext().

But for Leshan, perhaps we would rather like a 4.00 (Bad Request) with payload "Unauthorized" as response?

Reply:
Ok I will handle this by returning null in Leshan. 

> So for instance for an incoming request it will respond with a 4.01 (Unauthorized) error message with payload string "Security context not found". So if the OSCORE code in Californium is allowed to handle things, that will be the result of a null from getContext(). 

I quickly looked at [OSCORE RFC](https://datatracker.ietf.org/doc/html/rfc8613#section-8.2) and it says : 
_"If the server fails to retrieve a Recipient Context with Recipient ID corresponding to the 'kid' parameter received, the server MAY respond with a 4.01 (Unauthorized) error message."_

So I think it's the right behavior.

> But for Leshan, perhaps we would rather like a 4.00 (Bad Request) with payload "Unauthorized" as response?

Except we see something like this in the LWM2M specification, I would prefer to strictly respect the OSCORE RFC ?

Reply:
> I quickly looked at [OSCORE RFC](https://datatracker.ietf.org/doc/html/rfc8613#section-8.2) and it says : _"If the server fails to retrieve a Recipient Context with Recipient ID corresponding to the 'kid' parameter received, the server MAY respond with a 4.01 (Unauthorized) error message."_
> 
> So I think it's the right behavior.

Yes, this behaviour is based on the OSCORE RFC.

> Except we see something like this in the LWM2M specification, I would prefer to strictly respect the OSCORE RFC ?

Sure, that makes sense to me. Then we stick to the the current way it's done in the Californium OSCORE code, and returning null as you plan should result in the intended behaviour.

Reply:
(this should be done in 6c2b4c3)